README.md

```diff
@@ -677,28 +782,68 @@ module.exports = {
+You can use [Playwright's built in APIs](https://playwright.dev/docs/test-snapshots) for DOM snapshot testing:
 
 ```js
-export const MyStory = () => {
-  const isTestRunner = window.navigator.userAgent.match(/StorybookTestRunner/);
-  return (
-    <div>
-      <p>Is this story running in the test runner?</p>
-      <p>{isTestRunner ? 'Yes' : 'No'}</p>
-    </div>
-  );
+// .storybook/test-runner.js
+module.exports = {
+  async postRender(page, context) {
+    // the #root element wraps the story. From Storybook 7.0 onwards, the selector should be #storybook-root
+    const elementHandler = await page.$('#root');
+    const innerHTML = await elementHandler.innerHTML();
+    expect(innerHTML).toMatchSnapshot();
+  },
 };
 ```
 
-Given that this check is happening in the browser, it is only applicable in the following scenarios:
+When running with `--stories-json`, tests get generated in a temporary folder and snapshots get stored alongside. You will need to `--eject` and configure a custom [`snapshotResolver`](https://jestjs.io/docs/configuration#snapshotresolver-string) to store them elsewhere, e.g. in your working directory:
 
-- inside of a render/template function of a story
-- inside of a play function
-- inside of preview.js
-- inside any other code that is executed in the browser
+```js
+// ./test-runner-jest.config.js
+const path = require('path');
+
+module.exports = {
+  resolveSnapshotPath: (testPath, snapshotExtension) =>
+    path.join(process.cwd(), '__snapshots__', path.basename(testPath) + snapshotExtension),
+  resolveTestPath: (snapshotFilePath, snapshotExtension) =>
+    path.join(process.env.TEST_ROOT, path.basename(snapshotFilePath, snapshotExtension)),
+  testPathForConsistencyCheck: path.join(process.env.TEST_ROOT, 'example.test.js'),
+};
+```
+
+### Image snapshot
+
+Here's a slightly different recipe for image snapshot testing:
+
+```js
+// .storybook/test-runner.js
+const { waitForPageReady } = require('@storybook/test-runner');
+const { toMatchImageSnapshot } = require('jest-image-snapshot');
+
+const customSnapshotsDir = `${process.cwd()}/__snapshots__`;
+
+module.exports = {
+  setup() {
+    expect.extend({ toMatchImageSnapshot });
+  },
+  async postRender(page, context) {
+    // use the test-runner utility to wait for fonts to load, etc.
+    await waitForPageReady(page);
+
+    // If you want to take screenshot of multiple browsers, use
+    // page.context().browser().browserType().name() to get the browser name to prefix the file name
+    const image = await page.screenshot();
+    expect(image).toMatchImageSnapshot({
+      customSnapshotsDir,
+      customSnapshotIdentifier: context.id,
+    });
+  },
+};
+```
+
+There is also an exported `TestRunnerConfig` type available for TypeScript users.
```

Comment:
@yannbf, can't we turn this into a disclaimer section mentioning TypeScript support? I'm asking because when someone reads this and tries to look for the type or any other references to `TestRunnerConfig` we're only mentioning it with the `prepare` function which may confuse users on how to configure it and what's available. 

Let me know and we'll take it from there.

Reply:
I'll skip this part from now and come back to it once we upgrade Jest, which could improve out of the box Typescript support. Thanks!